File: javav2/example_code/glue/src/main/java/com/example/glue/ListWorkflows.java
//snippet-sourcedescription:[ListWorkflows.java demonstrates how to list workflows.]
//snippet-keyword:[AWS SDK for Java v2]
//snippet-keyword:[AWS Glue]
/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.example.glue;
//snippet-start:[glue.java2.list_wfs.import]
import software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider;
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.glue.GlueClient;
import software.amazon.awssdk.services.glue.model.GlueException;
import software.amazon.awssdk.services.glue.model.ListWorkflowsRequest;
import software.amazon.awssdk.services.glue.model.ListWorkflowsResponse;
import java.util.List;
//snippet-end:[glue.java2.list_wfs.import]

/**
 * Before running this Java V2 code example, set up your development environment, including your credentials.
 *
 * For more information, see the following documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */
public class ListWorkflows {

    public static void main(String[] args) {

        Region region = Region.US_EAST_1;
        GlueClient glueClient = GlueClient.builder()
            .region(region)
            .credentialsProvider(ProfileCredentialsProvider.create())
            .build();

        listAllWorkflows(glueClient);
        glueClient.close();
    }

    //snippet-start:[glue.java2.list_wfs.main]
    public static void listAllWorkflows( GlueClient glueClient) {

        try {
            ListWorkflowsRequest workflowsRequest = ListWorkflowsRequest.builder()
                .maxResults(10)
                .build();

            ListWorkflowsResponse workflowsResponse = glueClient.listWorkflows(workflowsRequest);
            List<String> workflows = workflowsResponse.workflows();
            for (String workflow: workflows) {
                System.out.println("Workflow name is: "+workflow);
            }

        } catch (GlueException e) {
            System.err.println(e.awsErrorDetails().errorMessage());
            System.exit(1);
        }
    }
   //snippet-end:[glue.java2.list_wfs.main]
}

File: kotlin/services/glue/src/main/kotlin/com/kotlin/glue/ListWorkflows.kt
// snippet-sourcedescription:[ListWorkflows.kt demonstrates how to list workflows.]
// snippet-keyword:[AWS SDK for Kotlin]
// snippet-keyword:[AWS Glue]
/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.glue

// snippet-start:[glue.kotlin.list_wfs.import]
import aws.sdk.kotlin.services.glue.GlueClient
import aws.sdk.kotlin.services.glue.model.ListWorkflowsRequest
// snippet-end:[glue.kotlin.list_wfs.import]

/**
Before running this Kotlin code example, set up your development environment,
including your credentials.

For more information, see the following documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */

suspend fun main() {
    listAllWorkflows()
}

// snippet-start:[glue.kotlin.list_wfs.main]
suspend fun listAllWorkflows() {

    val request = ListWorkflowsRequest {
        maxResults = 10
    }

    GlueClient { region = "us-east-1" }.use { glueClient ->
        val response = glueClient.listWorkflows(request)
        response.workflows?.forEach { workflow ->
            println("Workflow name is: $workflow")
        }
    }
}
// snippet-end:[glue.kotlin.list_wfs.main]
